xbmc/threads/test/TestHelpers.h

```diff
@@ -76,14 +76,8 @@ class thread
   inline thread(thread& other) : f(other.f), cthread(other.cthread) { other.f = NULL; other.cthread = NULL; }
   inline thread& operator=(thread& other) { f = other.f; other.f = NULL; cthread = other.cthread; other.cthread = NULL; return *this; }
 
-  void join()
-  {
-    cthread->Join(static_cast<unsigned int>(-1));
-  }
+  void join() { cthread->Join(std::chrono::milliseconds(-1)); }
```

Comment:
This looks wrong. Did you mean `std::chrono::milliseconds::max()`?

Reply:
I don't know, I guess the overflow causes a max?

Reply:
TBH I have no idea what `std::chrono::duration` does with negative values. As long as the comparison within the method matches... It's just odd that some places use `::max()` but here we use `-1`.

Reply:
Just saw that all predefined durations use a signed data type so this should be safe. Then only the question of consistancy remains.